[repo-agent] sigsegv in review sandbox

User: barney-s

When running the review, the sandbox had a SIGSEGV causing the pod to terminate. 

```
I0118 17:34:30.289953       1 review.go:513] "Filtering out invalid comment" file="extensions/controllers/sandboxwarmpool_controller.go" line=140
I0118 17:34:30.289959       1 review.go:521] "YAML validation successful."
I0118 17:34:30.289970       1 review.go:359] "Agent run and validation successful."
I0118 17:34:30.289990       1 codeserver.go:49] "Stopping code-server command"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x196a220]

goroutine 1 [running]:
github.com/gke-labs/gemini-for-kubernetes-development/repo-agent/pkg/commands.isDuplicateCommentExact(...)
        /app/pkg/commands/review.go:478
github.com/gke-labs/gemini-for-kubernetes-development/repo-agent/pkg/commands.(*ReviewCommand).Run(0xc00046a300, {0x21cc848, 0xc0003bdf40})
        /app/pkg/commands/review.go:374 +0x3c20
github.com/gke-labs/gemini-for-kubernetes-development/repo-agent/pkg/commands.(*ReviewDaemonCommand).Run(0xc00046a300, {0x21cc848, 0xc0003bdf40})
        /app/pkg/commands/review_daemon.go:63 +0x205
github.com/gke-labs/gemini-for-kubernetes-development/repo-agent/pkg/commands.BuildReviewDaemonCommand.func1(0xc000502608, {0x3287b00?, 0x4?, 0x1efcd03?})
        /app/pkg/commands/review_daemon.go:33 +0x69
github.com/spf13/cobra.(*Command).execute(0xc000502608, {0x3287b00, 0x0, 0x0})
        /go/pkg/mod/github.com/spf13/cobra@v1.9.1/command.go:1015 +0xaaa
github.com/spf13/cobra.(*Command).ExecuteC(0xc000271508)
        /go/pkg/mod/github.com/spf13/cobra@v1.9.1/command.go:1148 +0x46f
github.com/spf13/cobra.(*Command).Execute(...)
        /go/pkg/mod/github.com/spf13/cobra@v1.9.1/command.go:1071
github.com/spf13/cobra.(*Command).ExecuteContext(...)
        /go/pkg/mod/github.com/spf13/cobra@v1.9.1/command.go:1064
main.run({0x21cc848, 0xc0003bdf40})
        /app/cmd/repo-sandbox/main.go:65 +0x5d4
main.main()
        /app/cmd/repo-sandbox/main.go:21 +0x94
barni@barni.c ~/workspace/src/github.com/barney-s/gemini-for-kubernetes-development% kubectl logs -f -n barney-s devc-agent-sandbox-pr-240
Defaulted container "sandbox" out of: sandbox, gemini-configs (init), inject-agent (init)
I0118 17:34:31.065788       1 codeserver.go:36] "Starting code-server command"
I0118 17:34:31.066063       1 codeserver.go:49] starting code-server
```

[Screenshot attached in request but ignored due to missing image host configuration]

--- a/repo-agent/pkg/commands/review.go
+++ b/repo-agent/pkg/commands/review.go
@@ -372,7 +372,7 @@ func (c *ReviewCommand) Run(ctx context.Context) error {
 			accumulatedAgentOutput.Labels = uniqueStrings(accumulatedAgentOutput.Labels)
 		} else {
 			for _, newComment := range agentOutput.Review.Comments {
-				if newComment == nil {
+				if newComment == nil || newComment.Path == nil || newComment.Line == nil || newComment.Body == nil {
 					continue
 				}
 				if !isDuplicateCommentExact(newComment, existingComments, accumulatedAgentOutput.Review.Comments) {

